src/Forms/ValidatorList.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+namespace SilverStripe\Forms;
+
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\ValidationResult;
+
+/**
+ * Class ValidatorList
+ *
+ * @package SilverStripe\Forms
+ */
+class ValidatorList extends Validator
+{
+    /**
+     * @var ArrayList|Validator[]
+     */
+    private $validators;
+
+    public function __construct()
+    {
+        $this->validators = ArrayList::create();
+
+        parent::__construct();
+    }
+
+    /**
+     * @param Form $form
+     * @return Validator
+     */
+    public function setForm($form)
+    {
+        foreach ($this->getValidators() as $validator) {
+            $validator->setForm($form);
+        }
+
+        return parent::setForm($form);
+    }
+
+    /**
+     * Returns any errors there may be. This method considers the enabled status of the ValidatorList as a whole
+     * (exiting early if the List is disabled), as well as the enabled status of each individual Validator.
+     *
+     * @return ValidationResult
+     */
+    public function validate()
+    {
+        $this->resetResult();
+
+        // This ValidatorList has been disabled in full
+        if (!$this->getEnabled()) {
+            return $this->result;
+        }
+
+        $data = $this->form->getData();
+
+        foreach ($this->getValidators() as $validator) {
+            // Reset the validation results for this Validator
+            $validator->resetResult();
+
+            // This Validator has been disabled, so skip it
+            if (!$validator->getEnabled()) {
+                continue;
+            }
+
+            // Run validation, and exit early if it's valid
+            if ($validator->php($data)) {
+                continue;
+            }
+
+            // Validation result was invalid. Combine our ValidationResult messages
+            $this->getResult()->combineAnd($validator->getResult());
+        }
+
+        return $this->result;
+    }
+
+    /**
+     * Returns whether the field in question is required. This will usually display '*' next to the
+     * field.
+     *
+     * @param string $fieldName
+     *
+     * @return bool
+     */
+    public function fieldIsRequired($fieldName)
+    {
+        foreach ($this->getValidators() as $validator) {
+            if (!$validator->fieldIsRequired($fieldName)) {
+                continue;
+            }
+
+            return true;
```

Comment:
Could shorten this by reversing the condition 

```php
if ($validator->fieldIsRequired($fieldName)) {
    return true;
}
```
2 precious lines saved 😉

Reply:
These ones were more about sticking to "early exit". But, fair enough, they're pretty readable either way.